react/src/pages/analysis/Analysis.tsx

```diff
@@ -87,103 +89,93 @@ function createAnalysis(
+    createAnalysis(2, 'AN43820', '2020-06-19', 'User B', '4030', ['BB024', 'BB025', 'BB026'], 'CRG', '2hrs', PipelineStatus.RUNNING),
+    createAnalysis(3, 'AN38292', '2020-06-22', 'User A', '3291', ['AA810', 'AA811', 'AA812', 'AA813'], 'CRG', '2hrs', PipelineStatus.RUNNING),
+    createAnalysis(4, 'AN33889', '2020-06-21', 'User C', '3289', ['CC330'], 'CRE', '17hrs', PipelineStatus.RUNNING),
+    createAnalysis(5, 'AN38920', '2020-06-21', 'User A', '2382', ['AC289', 'AC290', 'AC291'], 'CRG', '20hrs', PipelineStatus.RUNNING),
+    createAnalysis(6, 'AN38921', '2020-06-19', 'User B', '4182', ['AA337', 'AA338', 'AA339'], 'CRG', '47hrs', PipelineStatus.COMPLETED),
+    createAnalysis(7, 'AN38991', '2020-06-19', 'User B', '3271', ['AA320'], 'CRE', '20hrs', PipelineStatus.COMPLETED),
+    createAnalysis(8, 'AN20032', '2020-06-20', 'User C', '3839', ['CC773', 'CC774', 'CC775'], 'CRE', '22hrs', PipelineStatus.ERROR),
 ];
 
 export default function Analysis() {
     const classes = useStyles();
-    const [open, setOpen] = useState(false);
+    const [detail, setDetail] = useState(false);
+    const [cancel, setCancel] = useState(false);
     const [activeRow, setActiveRow] = useState<AnalysisRun | null>(null);
 
     useEffect(() => {
         document.title = "Analyses | ST2020";
     }, []);
 
-    const formatDialog = (row: AnalysisRun) => {
-        return (
-            <div>
-                <DialogTitle className={classes.dialogHeader}>Analysis {row.analysisID}</DialogTitle>
-                <DialogContent>
-                    <div className={classes.statusTitle}><b>{row.status}</b></div>
-                    <Grid container spacing={0}>
-                        <Grid item xs={12}>
-                            <b>Project: </b>{row.project}
-                        </Grid>
-                        <Grid item xs={12}>
-                            <b>Date Submitted: </b>{row.dateSubmitted}
-                        </Grid>
-                    </Grid>
-                </DialogContent>
-            </div>
-        )
-    }
-
     return (
         <main className={classes.content}>
             <div className={classes.appBarSpacer} />
+            {(activeRow as AnalysisRun) && 
+            <CancelAnalysisDialog
+                open={cancel}
+                title={`Stop Analysis ${(activeRow as AnalysisRun).analysisID}?`}
+                message={`Do you really want to stop the analysis of samples ${(activeRow as AnalysisRun).participants.join(', ')}? Stopping an analysis will delete all intermediate files and progress. Input files will remain untouched.`}
+                onClose={() => { setCancel(false) }}
+            />}
+            {(activeRow as AnalysisRun) &&
+            <AnalysisInfoDialog
+                open={detail}
+                analysis={activeRow as AnalysisRun}
+                onClose={() => setDetail(false)}
+            />}
```

Comment:
The type casts in the above block are unnecessary.

Reply:
Type cast is needed, otherwise I get this error since activeRow is of type AnalysisRun | null:

![Screen Shot 2020-07-07 at 1 48 41 PM](https://user-images.githubusercontent.com/18020674/86821678-b1270180-c058-11ea-94b4-2bfbe1cb52ca.png)

Reply:
That's strange, I'm not getting then because with `activeRow &&` it implies that for the rest of the expression `activeRow` cannot be `null`...